### PR TITLE
[ALLUXIO-1950] Improve cleanup of test cluster directories

### PR DESCRIPTION
**Pull request by aaudiber**
https://alluxio.atlassian.net/browse/ALLUXIO-1950

Instead of creating temp directories in /tmp, we will create them in /tmp/alluxio-tests/ and delete this directory at the start and end of testing.

### CODE REVIEW

File: core/common/src/test/java/alluxio/CommonTestUtils.java [hunk 1]
```diff
@@ -190,4 +199,60 @@ public static void waitFor(String description, Function<Void, Boolean> condition
...
+    }
+    return sAlluxioTestDirectory;
+  }
+
+  /**
+   * Creates the Alluxio testing directory, deleting the previous one if it still exists.
+   *
+   * @return the directory
+   */
+  private static File createTestingDirectory() {
```
Comment (yupeng9) [inline core/common/src/test/java/alluxio/CommonTestUtils.java]: Can we call this on `sAlluxioTestDirectory` declaration?

**Comment by aaudiber** [comment]
Jenkins, test this please

File: core/common/src/test/java/alluxio/CommonTestUtils.java (hunk 1 above)
Comment (aaudiber) [inline core/common/src/test/java/alluxio/CommonTestUtils.java]: Done

**Comment by yupeng9** [comment]
LGTM

File: core/common/src/test/java/alluxio/CommonTestUtils.java [hunk 2]
```diff
@@ -190,4 +199,57 @@ public static void waitFor(String description, Function<Void, Boolean> condition
     }
     return fields;
   }
+
+  /**
+   * @return the directory to be used for temporary files in Alluxio tests
+   */
+  public static synchronized File getAlluxioTestDirectory() {
```
Comment (gpang) [inline core/common/src/test/java/alluxio/CommonTestUtils.java]: Is this intentionally `public`? Can the variable be accessed directly? Nothing external calls this...

Comment (aaudiber) [inline core/common/src/test/java/alluxio/CommonTestUtils.java]: I've exposed the variable directly and removed this method, thanks

**Comment by aaudiber** [comment]
@gpang Comments addressed, PTAL

**Comment by gpang** [comment]
LGTM

**Pull request merged**